packages/@glimmer/runtime/lib/tags.ts

```diff
@@ -0,0 +1,30 @@
+import { Slice, LinkedListNode } from '@glimmer/util';
+import { Tag, Tagged, createCombinatorTag, CONSTANT_TAG } from '@glimmer/tag';
+
+export function combineTagged(tagged: ReadonlyArray<Tagged>): Tag {
```

Comment:
Why is this in runtime and not tag?

Reply:
I wanted to avoid pulling extra types into `@glimmer/tag` for things that are essentially unrelated. These combinators are really optimizations for a couple of narrow use cases in the VM itself, and while I wasn't comfortable removing them/simplifying them without perf tests, I also don't think they really belong with the rest of the tracking infra, conceptually.

If perf wasn't a concern, or tests showed it wasn't an issue, I probably would have made them use the `combine()` function like everyone else. If perf is an issue, I think the internal `createCombinatorTag` API is good for solving the problem.